roles/matrix-backup-borg/tasks/validate_config.yml

```diff
@@ -7,4 +7,9 @@
   with_items:
     - "matrix_backup_borg_ssh_key_private"
     - "matrix_backup_borg_location_repositories"
-    - "matrix_backup_borg_storage_encryption_passphrase"
+
+- name: Fail if encryption passphrase is undefined unless repository is unencrypted
+  fail:
+    msg: >-
+      You need to define a required passphrase.
+  when: "matrix_backup_borg_storage_encryption_passphrase == '' and not matrix_backup_borg_encryption == 'none'"
```

Comment:
So, to enable the usage of non-encrypted repositories, one will need both `matrix_backup_borg_unknown_unencrypted_repo_access_is_ok: true` and `matrix_backup_borg_encryption: 'none'`?

This makes the comment above the `matrix_backup_borg_encryption` definition in `defaults/main.yml` incorrect:

> # borg encryption mode, only repokey-* is supported
> matrix_backup_borg_encryption: repokey-blake2

If so, I suppose we can change the default definition for `matrix_backup_borg_unknown_unencrypted_repo_access_is_ok` to be like this:

```yaml
matrix_backup_borg_unknown_unencrypted_repo_access_is_ok: "{{ matrix_backup_borg_encryption == 'none' }}"
```

Then, to enable unencrypted backup usage, we'll only need `matrix_backup_borg_encryption: 'none'`.

-----

Additionally, this `when` statement here (`not matrix_backup_borg_encryption == 'none'`) is confusing and can be done like this `matrix_backup_borg_encryption != 'none'`.

------

Also, `'none'` (a string) can be confused with `none` (`null`).

Perhaps what you mean is `matrix_backup_borg_encryption is none` or `matrix_backup_borg_encryption is not none`, instead of comparing to a string.

Reply:
> Also, `'none'` (a string) can be confused with `none` (`null`).
> 
> Perhaps what you mean is `matrix_backup_borg_encryption is none` or `matrix_backup_borg_encryption is not none`, instead of comparing to a string.

The borgmatic config uses this term, should I change it to something else?